### PR TITLE
Fix _obtain_metadata in Profile class

### PR DESCRIPTION
**Pull request by kunwar31**
Fixes errors in instaloader.Profile.from_username,  instaloader.Instaloader().check_profile_id where ProfileNotFoundException is raised by KeyError on graphql

### CODE REVIEW

File: instaloader/structures.py [hunk 1]
```diff
@@ -569,8 +569,8 @@ def _asdict(self):
     def _obtain_metadata(self):
         try:
             if not self._has_full_metadata:
-                metadata = self._context.get_json('{}/'.format(self.username), params={})
-                self._node = metadata['entry_data']['ProfilePage'][0]['graphql']['user']
+                metadata = self._context.get_json('{}/'.format(self.username), params={'__a': 1})
+                self._node = metadata['graphql']['user']
                 self._has_full_metadata = True
                 self._rhx_gis = metadata.get('rhx_gis')
```
Comment (aandergr) [inline instaloader/structures.py:575]: `self._rhx_gis` is needed for several queries. With your change, `rhx_gis` is not in `metadata` anymore, leading `self._rhx_gis` to be set to `None`. Do `Profile.get_*_posts()` still work after this change?

Comment (kunwar31) [reply instaloader/structures.py:575]: @aandergr Profile.get_*_posts() is still working. I'm not familiar with what "rhx_gis" is required for, as it looks like Instagram has deprecated it.  If it is required for some endpoints which I'm not aware of, we should request to this endpoint twice, with and without "__a": 1 parameter, and check which result contains required data.

**Comment by kunwar31** [comment]
> Thanks for your Pull Request. What is it for? Does it address any reported bug?

Essentially #297, but keyError "graphql" , ProfilePage list in entry_data key is now an empty list, which raises this keyError

**Comment by tskho** [comment]
> Essentially #297, but keyError "graphql" , ProfilePage list in entry_data key is now a list with an empty dictionary, which raises this keyError

Also fixed 429 errors for me. Instaloader is operational again. Kudos

**Comment by aandergr** [comment]
> > Thanks for your Pull Request. What is it for? Does it address any reported bug?
> 
> Essentially #297, but keyError "graphql" , ProfilePage list in entry_data key is now a list with an empty dictionary, which raises this keyError

That makes no sense, #297 has been closed a year ago and the code does not look like it could fix a KeyError accessing 'rhx_gis'.

**Comment by kunwar31** [comment]
@aandergr you're right, I meant similar to #297. I should have created a separate issue for this, as it's very different from #297 as I mentioned before -  "keyError "graphql" , ProfilePage list in entry_data key is now a list with an empty dictionary, which raises this keyError". 
Instead I have directly created the pull request to fix the issue, rather than creating/explaining the issue first.

**Comment by kunwar31** [comment]
This change also fixes #643 #654 , as it is related to broken _obtain_metadata in Profile

**Comment by kunwar31** [comment]
I don't know how many issues will be fixed by this I haven't went through all the issues, I'm able to find more as I go through them #641

**Comment by kunwar31** [comment]
#563 as well

**Comment by cloudrac3r** [comment]
This does NOT fix 643 or 654. With `pip3 install instaloader`, [I get this output,](https://cadence.moe/pastes/6832) and with your pull request, [I get this.](https://cadence.moe/pastes/6833) TLDR, redirected to the login page in both cases.

In both cases the server is just blocked by Instagram:

```
$ curl https://www.instagram.com/instagram/ -I -X GET
HTTP/2 302 
location: https://www.instagram.com/accounts/login/?next=/instagram/
[...]
```

This block is _not_ relieved by waiting 11 minutes. It has been in effect since around June 1st.

On my computer at home, which isn't blocked by Instagram, both instaloader from `pip3 install instaloader` and the branch of this pull request work.

> Also fixed 429 errors for me. Instaloader is operational again. Kudos

@tskho — I expect that it fixed those errors because the block expired in the time gap between using the regular release and using this pull request. If you can, please briefly test with this pull request again, and if it is able to download, stop it and test using the regular release, and tell me if the regular release works.

**Comment by kunwar31** [comment]
@cloudrac3r Without logging in, both endpoints don't work. With login, __a=1 works for ~500 profiles, and then blocks for a while. Can you try with login?

**Comment by cloudrac3r** [comment]
I'm sorry, I don't wish to create an account. If you have a spare account that you use for testing, and it's safe to share credentials, feel free to direct those [to one of my contact addresses](https://cadence.moe/about/contact) and I'll use it to test ASAP.

**Comment by aandergr** [comment]
Many of the here-mentioned supposedly fixed issues are already closed, are not even bugs, or do not seem in any way related to the proposed change. That makes no sense. We cannot introduce a so weakly legitimated change to Instaloader.

**Comment by tskho** [comment]
> @tskho — I expect that it fixed those errors because the block expired in the time gap between using the regular release and using this pull request. If you can, please briefly test with this pull request again, and if it is able to download, stop it and test using the regular release, and tell me if the regular release works.

I cannot comment as to why this pull request works or if code is of acceptable quality but here's what I did:

``` 
$ pipenv --three install git+https://github.com/instaloader/instaloader.git#egg=instaloader
$ pipenv run instaloader --login=vtskho vtskho
Loaded session from /home/vitaly/.config/instaloader/session-vtskho.
Logged in as vtskho.
Profile vtskho does not exist.
The most similar profiles are: vtskho, vtskhovrebova, seniora_d_z_l_01, inessashekhovtseva, bykhovtsevakairova.
Trying again anonymously, helps in case you are just blocked.

HTTP redirect from https://www.instagram.com/vtskho/ to https://www.instagram.com/accounts/login/?next=/vtskho/
JSON Query to vtskho/: 429 Too Many Requests: redirected to login [retrying; skip with ^C]
HTTP error code 429 was returned because too many queries occurred in
the last time. Please do not use Instagram in your browser or run
multiple instances of Instaloader in parallel.
The request will be retried in 666 seconds, at 13:21.
GraphQL requests:
  other
    last 10 minutes: 1 requests
    last 11 minutes: 1 requests
    last 15 minutes: 1 requests
    last 20 minutes: 1 requests
    last 30 minutes: 1 requests
    last 60 minutes: 1 requests
```

and then right after that, using suggested pull request, fresh environment in different directory:

```
$ pipenv install git+https://github.com/kunwar31/instaloader.git#egg=instaloader
$ pipenv run instaloader --login=vtskho vtskho
Loaded session from /tmp/.instaloader-vitaly/session-vtskho.
Logged in as vtskho.
Stored ID 2138927824 for profile vtskho.
vtskho/2018-11-21_19-35-46_UTC_profile_pic.jpg
Retrieving posts from profile vtskho.
[ 1/20] vtskho/2020-06-03_20-26-23_UTC.jpg [Джунгли зовут!  #lucky to not…] json
[ 2/20] vtskho/2020-02-14_18-47-16_UTC.jpg [Well, hello] json
[ 3/20] vtskho/2020-01-13_11-41-48_UTC.jpg [Тяжелые отношения с Прагой. С…] json
[ 4/20] vtskho/2019-12-20_04-19-23_UTC.jpg [Americans need a little bit o…] json
[ 5/20] vtskho/2019-11-16_19-20-46_UTC.jpg [This tiny piece of paper stil…] json
[ 6/20] ^C
Interrupted by user.
```

Functionality difference is obvious to me. Perhaps there's something useful in this pull request?

**Comment by cloudrac3r** [comment]
Huh. Thanks for that log.

**Comment by kunwar31** [comment]
@tskho Thank you for testing and providing evidence for the proposed functionality

**Pull request closed without merging**